clang/lib/CodeGen/CGOpenCLRuntime.cpp

```diff
@@ -37,6 +37,28 @@ llvm::Type *CGOpenCLRuntime::convertOpenCLSpecificType(const Type *T) {
   llvm::LLVMContext& Ctx = CGM.getLLVMContext();
   uint32_t AddrSpc = CGM.getContext().getTargetAddressSpace(
       CGM.getContext().getOpenCLTypeAddrSpace(T));
+
+  if (CGM.getTriple().isNVPTX()) {
+    switch (cast<BuiltinType>(T)->getKind()) {
+    default:
+      break;
+#define IMAGE_TYPE(ImgType, Id, SingletonId, Access, Suffix)                   \
+  case BuiltinType::Id:                                                        \
+    return llvm::IntegerType::getInt64Ty(CGM.getLLVMContext());
+#include "clang/Basic/OpenCLImageTypes.def"
+#define IMAGE_TYPE(ImgType, Id, SingletonId, Access, Suffix)                   \
```

Comment:
Does this need to #undef-ed before being re-defined?

Reply:
`OpenCLImageTypes.def` performs the `#undef IMAGE_TYPE`